backend/src/packages/follow/follow.repository.ts

```diff
@@ -0,0 +1,65 @@
+import { type FollowModel } from './follow.model.js';
+import { type UserFollowResponseDto } from './libs/types/types.js';
+
+class FollowRepository {
+  private followModel: typeof FollowModel;
+
+  public constructor(followModel: typeof FollowModel) {
+    this.followModel = followModel;
+  }
+
+  public async toggleFollowAuthor({
+    userId,
+    authorId,
+  }: {
+    userId: number;
+    authorId: number;
+  }): Promise<FollowModel | FollowModel[]> {
+    const isFollowing = await this.followModel
+      .query()
+      .where({ userId, authorId })
+      .first();
+
+    if (!isFollowing) {
+      return await this.followModel
+        .query()
+        .insert({ userId, authorId })
+        .returning('*');
+    }
+
+    return await this.followModel
+      .query()
+      .delete()
+      .where({ userId, authorId })
+      .returning('*');
+  }
+
+  public async getAuthorFollowersCountAndStatus({
+    userId,
+    authorId,
+  }: {
+    userId?: number;
+    authorId: number;
+  }): Promise<UserFollowResponseDto> {
+    return await this.followModel.transaction(async (trx) => {
+      const followersCount = await this.followModel
+        .query(trx)
+        .where({ authorId })
+        .count()
+        .first()
+        .castTo<{ count: string }>();
+
+      const isFollowed = userId
+        ? await this.followModel.query(trx).where({ userId, authorId }).first()
+        : false;
+
+      return {
+        authorId,
+        followersCount: Number.parseInt(followersCount?.count, 10),
```

Comment:
Can we cast it this way?
```suggestion
        followersCount: Number(followersCount?.count ?? 0),
```

Reply:
@andanf-e 
We can do it this way:
it should return 0 anyway if there is nothing
```suggestion
        followersCount: Number(followersCount.count),
```

Reply:
[c376cc8](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/280/commits/c376cc8dc18ae8989f577807fafaada40071ae37)